lxd/storage/backend_lxd.go

```diff
@@ -7560,6 +7560,49 @@ func (b *lxdBackend) ListUnknownVolumes(op *operations.Operation) (map[string][]
 // the state of the volume on disk, and if all checks out, it adds the parsed backup file contents to projectVols.
 // In any case it also checks whether or not the instance has unknown custom volumes attached and appends them to projectVols too.
 func (b *lxdBackend) detectUnknownInstanceAndCustomVolumes(vol *drivers.Volume, projectVols map[string][]*backupConfig.Config, op *operations.Operation) error {
+	volUUID := vol.Config()["volatile.uuid"]
+
+	// In case a driver is using volumes with transformed names, we have to identify which instance in LXD
```

Comment:
This feels like the wrong place to do this task (also this function is already too big so adding more to it is not attractive).


Lets think more about phases in the caller of this function:

1. Ask the storage driver for volumes - and as we know, for powerflex, pure and hpe, the volume names they return will be uuids not the actual volume name we want in the DB.
2. The missing step: We need to transform those volumes (if we can) into volume structs acceptable for the rest of the system (and potentially remove ones that aren't compatible, such as custom volumes, although im not sure if that part isn't better done in the storage driver).
3. Call `detectUnknownInstanceAndCustomVolumes` passing in the transformed volume struct.